Lesson4/src/com/home/model/fabric/Person/PersonFabric.java

```diff
@@ -0,0 +1,33 @@
+package com.home.model.fabric.Person;
+
+import com.home.exception.MyNFException;
+import com.home.model.Address;
+import com.home.model.Person;
+import com.home.model.fabric.Fabric;
+
+
+import java.util.List;
+
+public class PersonFabric implements Fabric<Person> {
+    @Override
+    public Person getSomeObject(List<String> initData) {
+
+        if (initData.size() == 0) {
+            System.err.println("Wrong input! Empty input!");
+            return null;
+        }
+        String[] formattedData = initData.get(0).split(" ");
+        if (formattedData.length < 6) {
+            System.err.println("Wrong input! Need more data!");
+            return null;
+        }
+        try {
+            Person person = new Person(Integer.parseInt(formattedData[0]), formattedData[1], Integer.parseInt(formattedData[2]), formattedData[3].toCharArray()[0]);
+            Address address = new Address(formattedData[4], formattedData[5]);
+            person.setAddress(address);
+            return person;
+        } catch (NumberFormatException e) {
+            throw new MyNFException(e);
```

Comment:
I see that you are throwing this exception, but there are no places where you catch it, please fix it

I also wonder if you actually need to catch number format exception here and throw yours instead. Please explain me your idea

Reply:
I looked my code and understand what my exception need catch earlear and NumberFormatException in Fabrics is better choice instead my custom exception. Remade realization again :)